samples/NameFormatting.js

```diff
@@ -0,0 +1,140 @@
+                {fit: true}
+            ]},
+            {tag: "br"},
+            
+            {content: rb.getString("Length"), classes: "ilib-onyx-sample-divider"},
+            {kind: "onyx.RadioGroup", name: "length", onActivate: "buttonActivated", components: [
+                {content: "Short"},
+                {content: "Medium"},
+                {content: "Long"}
+            ]},
+            
+            {content: rb.getString("(or) Parts"), classes: "ilib-onyx-sample-divider"},
+	    {classes: "namepart", components: [
+		{kind:"enyo.Checkbox", name: "prefixCbox", content: "Prefix ", onchange:"checkboxChanged"},
+ 		{kind:"enyo.Checkbox", name: "givenCbox",  content: "Given ",  onchange:"checkboxChanged"},
+		{kind:"enyo.Checkbox", name: "middleCbox", content: "Middle ", onchange:"checkboxChanged"},
+		{kind:"enyo.Checkbox", name: "familyCbox", content: "Family ",  onchange:"checkboxChanged"},
+		{kind:"enyo.Checkbox", name: "suffixCbox", content: "Suffix ",  onchange:"checkboxChanged"}
+	    ]},
+
+	    {components: [
+                {content: rb.getString("Prefix"), classes: "ilib-onyx-sample-divider"},
+                    {kind: "onyx.InputDecorator", alwaysLooksFocused: true, components: [
+                        {kind: "onyx.Input", name: "prefixInput", placeholder: rb.getString("Prefix")}
+                ]},
+		{content: rb.getString("Given Name"), classes: "ilib-onyx-sample-divider"},
+                    {kind: "onyx.InputDecorator", alwaysLooksFocused: true, components: [
+                        {kind: "onyx.Input", name: "givenInput", placeholder: rb.getString("Given Name")}
+                ]},
+		{content: rb.getString("Middle Name"), classes: "ilib-onyx-sample-divider"},
+                    {kind: "onyx.InputDecorator", alwaysLooksFocused: true, components: [
+                        {kind: "onyx.Input", name: "middleInput", placeholder: rb.getString("Middle Name")}
+                ]},
+		{content: rb.getString("Family Name"), classes: "ilib-onyx-sample-divider"},
+                    {kind: "onyx.InputDecorator", alwaysLooksFocused: true, components: [
+                        {kind: "onyx.Input", name: "familyInput", placeholder: rb.getString("Family Name")}
+                ]},
+		{content: rb.getString("Suffix"), classes: "ilib-onyx-sample-divider"},
+                    {kind: "onyx.InputDecorator", alwaysLooksFocused: true, components: [
+                        {kind: "onyx.Input", name: "suffixInput", placeholder: rb.getString("Suffix")}
+                ]}
+            ]}
+        ]},
+        {tag: "br"},
+        {kind: "onyx.Groupbox", classes:"onyx-sample-result-box", components: [
+            {kind: "onyx.GroupboxHeader", content: rb.getString("Format result:")},
+            {name: "rtlResult", fit: true, content: "-", style: "padding: 10px"}
+        ]}
+    ],
+ 
+    buttonActivated: function(inSender, inEvent) {
+	
+	if (this.$.length.getActive().content === 'Short') {
```

Comment:
This needs to be guarded (or one of the length items needs to be active by default) as this currently results in an exception with no length item being selected.